### PR TITLE
Use chat template

### PR DESCRIPTION
**Pull request by DePasqualeOrg**
Since https://github.com/huggingface/swift-transformers/pull/104, it's now possible to use the chat template from `tokenizer_config.json`. ~~I've updated LLMEval to use the chat template, but I noticed that the output from the Mistral models ends with `<|im_end|>`, even when this string is included in `extraEOSTokens`. Perhaps @pcuenca has an idea why this might be happening?~~

### CODE REVIEW

**Comment by DePasqualeOrg** [comment]
Using the chat templates with Llama 3.2 will result in an error until `none` and `tojson` are implemented in the Jinja package: https://github.com/maiqingqiang/Jinja/pull/4

File: Libraries/LLM/Models.swift
Comment (davidkoski) [inline Libraries/LLM/Models.swift:134]: This doesn't remove it from the output -- it tells it to stop producing tokens if it is seen.

**Comment by davidkoski** [comment]
> Using the chat templates with Llama 3.2 will result in an error until `none` and `tojson` are implemented in the Jinja package: [maiqingqiang/Jinja#4](https://github.com/maiqingqiang/Jinja/pull/4)

OK, should we hold off on taking this until that is merged (and we can pick it up here)?

File: Libraries/LLM/Models.swift
Comment (DePasqualeOrg) [reply Libraries/LLM/Models.swift:134]: I just don't understand why this token is now appearing at the end of the output when the chat template is used, which wasn't the case before. Are applications expected to remove it? I remember another model (maybe Gemma 2) that had a similar issue with the output, but I can't find it at the moment.

**Comment by DePasqualeOrg** [comment]
> OK, should we hold off on taking this until that is merged (and we can pick it up here)?

Yes, since LLMEval includes model configs for Llama 3.2. In the meantime, this branch can be used to test Llama 3.2 with the work in progress on Jinja.

**Comment by DePasqualeOrg** [comment]
We had a similar issue with an EOS token at the end of the output of Gemma 2: https://github.com/ml-explore/mlx-swift-examples/issues/89

This shouldn't be appearing in the output if it's included in `extraEOSTokens`. Maybe it's an issue with the tokenizer?

**Comment by DePasqualeOrg** [comment]
Llama 3.2 1B and 3B and Mistral 7B now work with chat templates, thanks to the changes in swift-transformers and Jinja.

**Review by davidkoski — APPROVED** [review]
Looks great, thank you!

**Pull request merged**
